webview-ui/test-generation/src/state.ts

```diff
@@ -360,6 +365,26 @@ export const reducer = (current: State, action: Action): State => {
         },
       };
     }
+    case 'loadAppNames': {
+      let apps = action.value;
+      apps = apps.filter(
+        (item) =>
+          item.metadata === null ||
+          !('is_simulator' in item.metadata) ||
+          item.metadata.is_simulator === false,
+      );
+      apps.forEach((app) => {
+        app.kind = app.kind === 'android' ? 'Android' : 'iOS';
```

Comment:
See my other comments w/r/t/ massaging the data at the HTTP client level or type guarding.